sklearn/tests/test_base.py

```diff
@@ -436,6 +436,7 @@ def __init__(self, attribute_pickled=5):
 
     def __getstate__(self):
         data = self.__dict__.copy()
+        data["_sklearn_version"] = sklearn.__version__
```

Comment:
I still think that `BaseEstimator` shouldn't overwrite `__getstate__` and scikit-learn should probably just not try to do those version checks. I understand the intention of course, but basically scikit-learn user's must have a strategy in place for making sure that they don't unpickle estimators with the wrong scikit-learn version. To me seems scikit-learn tries here to address a general problem with `pickle` in a very scikit-learn specific way, which has unintended side-effects on user code that ran fine for years.

To really address the problem of serialization-versioning would probably mean to introduce special serialization code (which would mean A LOT of code writing). But then, you could start to maintain a versionizable serialization interface.

(@zxcvbnius this comment is not about your code in this PR, but about the background of the code you modify).